src/components/controls/attachments/AddAttachmentPopup.tsx

```diff
@@ -1,82 +1,84 @@
-import React, {Component} from 'react';
+import React, {useState} from 'react';
 import {connect} from 'react-redux';
 import {FormLabel, FormGroup, Alert} from 'react-bootstrap';
+
 import {t} from '../../utils';
 import {Attachment} from '../../../models';
 import {ConfacState} from '../../../reducers/app-state';
-import {AttachmentDropzone} from './AttachmentDropzone';
 import {PopupButton, Popup} from '../Popup';
 import {SimpleCreatableSelect} from '../form-controls/select/SimpleCreatableSelect';
+import {AddIcon} from '../Icon';
+import {AttachmentDropzone} from './AttachmentDropzone';
 
 
 type AddAttachmentPopupProps = {
   attachments: Attachment[],
-  onClose: (...args: any[]) => any,
   onAdd: ({file: File, type: string}) => void,
   attachmentTypes: string[],
-  isOpen: boolean,
 }
 
-type AddAttachmentPopupState = {
-  type: string,
-  file: File | null,
-}
+const _AddAttachmentPopup = (props: AddAttachmentPopupProps) => {
+  const [file, setFile] = useState<File | null>(null);
+  const [type, setType] = useState<string>('');
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
 
-class AddAttachmentPopupComponent extends Component<AddAttachmentPopupProps, AddAttachmentPopupState> {
-  constructor(props: AddAttachmentPopupProps) {
-    super(props);
-    this.state = {
-      type: '',
-      file: null,
-    };
-  }
+  const _onUpload = () => {
+    props.onAdd({type, file});
+    setIsModalOpen(false);
+  };
 
-  _onUpload() {
-    this.props.onAdd(this.state);
-    this.props.onClose();
-  }
+  const {attachments} = props;
+  const canAddFile = type && !attachments.map(a => a.type.toUpperCase()).includes(type.toUpperCase());
 
-  render() {
-    if (!this.props.isOpen) {
-      return null;
-    }
+  const buttons: PopupButton[] = [{
+    text: t('cancel'),
+    onClick: () => setIsModalOpen(false),
+    variant: 'light',
+  }, {
+    text: t('add'),
+    variant: 'primary',
+    onClick: _onUpload,
+    disabled: !canAddFile || !file,
+  }];
 
-    const {attachments, onClose} = this.props;
-    const currentType = this.state.type;
-    const canAdd = currentType && !attachments.map(a => a.type.toUpperCase()).includes(currentType.toUpperCase());
+  return (
+    <>
+      <AddIcon
+        style={{marginTop: 0, marginLeft: 16, marginBottom: 26}}
+        onClick={() => setIsModalOpen(true)}
+        label={t('invoice.attachmentsAdd')}
+        size={1}
+        data-tst="add-attachment"
```

Comment:
Ter info: vergeet alle data-tst maar: je mag overnemen als het bestaat, of deleten als het bestaat maar je hoeft het niet toe te voegen als het er nog niet is.

--> Die data-tst zijn er gekomen omdat iemand het project ging gebruiken om selenium testing te leren.
Maar dat is (natuurlijk) nooit effectief gebeurt.

En ik zou de data-tst nu op een minder invasieve manier opnieuw implementeren.